smartapps/samsung/sleepsense-fall-asleep.src/sleepsense-fall-asleep.groovy

```diff
@@ -0,0 +1,151 @@
+/**
+ *  SLEEPsense-fall asleep
+ *
+ *  Copyright 2015 Samsung Mobileapp
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+definition(
+    name: "SLEEPsense-fall asleep",
+    namespace: "samsung",
+    author: "Samsung Mobileapp",
+    description: "When you fall asleep, do this.",
+    category: "My Apps",
+    iconUrl: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience.png",
+    iconX2Url: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience@2x.png",
+    iconX3Url: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience@2x.png")
+
+
+
+preferences {
+	page(name:"Mainpage", install:true, uninstall:true) {
+        section("When you fall asleep, do this.") {
+            input "sleepsensor", "capability.sleepSensor", title: "Sleep Sensor", multiple: false, required: true
+        }
+       section("Select lights/bulbs") {
+            input "LightLevelController", "capability.switchLevel", title: "Select light", multiple: true, required: false
+            input(name: "lightlevel", type: "enum", title: "level", options: ["off", "10%","20%","30%","40%","50%","60%","70%","80%","90%","100%"])
+        }
+	}
+}
+
+
+def installed()
+{
+	log.trace "installed()"
+   	state.sleepAction = "Sleeping"
+
+    subscribe(sleepsensor, "sleeping", sleepsensorHandler)    
+}
+
+def updated()
+{
+	log.trace "updated()"
+	state.sleepAction = "Sleeping"
+    
+	unsubscribe()
+    subscribe(sleepsensor, "sleeping", sleepsensorHandler)
+    subscribe(sleepsensor, "bedstate", sleepsensorHandler)
+  
+}
+
+def initialize() {
+	// TODO: subscribe to attributes, devices, locations, etc.
+}
+
+
+def sleepsensorHandler(evt) {
+	//log.debug "sleepsensorHandler: $evt.name"	
+    log.debug "sleepsensorHandler: $evt.value"
+    state.isOK = false;
+    
+    if((state.sleepAction == "Sleeping") && (evt.value == "sleeping"))
+    {
+    	log.debug "Sleeping actions will be excuted"
+    	state.isOK = true;
+    }
+    else
+    {
+    	return
+    }
+    
+    LightLevelConv()
+    LightLevelController()
+   
+}
+
+def LightLevelConv() {
+	if (lightlevel == "off") {
```

Comment:
consider using a switch statement instead of a bunch of conditional if/else statements